fava/templates/_tree_table.html

```diff
@@ -178,3 +178,41 @@
     </li>
 </ol>
 {% endmacro %}
+
+{% macro net_profit(income, expenses) %}
```

Comment:
This seems like a lot of duplicated code - I think it would be better to reuse the normal tree-table for this (you can create a tree containing a single "Net Profit" account for that)

Reply:
Thanks for the tip about tree/treenode, I've given it a try in the new version of the PR. There is no template code duplication remaining, but I had to add an extra method to the ledger object, in order to have access to the tree constructor (and given I had to do that, I've put there the logic of computing the net profit).